modules/siddhi-extensions/time/src/test/java/org/wso2/siddhi/extension/time/DateFormatFunctionExtensionTestCase.java

```diff
@@ -83,4 +84,230 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
         Assert.assertTrue(eventArrived);
         executionPlanRuntime.shutdown();
     }
+    @Test
+    public void dateFormatFunctionExtension2() throws InterruptedException {
+
+        log.info("DateFormatFunctionExtensionInvalidFormatTestCase");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (symbol string," +
+                "dateValue string,sourceFormat string,timestampInMilliseconds long,targetFormat string);";
+        String query = ("@info(name = 'query1') from inputStream select symbol , " +
+                "time:dateFormat(dateValue,targetFormat,sourceFormat) as formattedDate," +
+                "time:dateFormat(timestampInMilliseconds," +
+                "targetFormat) as formattedUnixDate insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager
+                .createExecutionPlanRuntime(inStreamDefinition + query);
+
+        executionPlanRuntime.addCallback("query1", new QueryCallback() {
+
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+
+                eventArrived = true;
+                for(int cnt=0;cnt<inEvents.length;cnt++){
+                    count++;
+                    log.info("Event : " + count + ",formattedDate : " + inEvents[cnt].getData(1) + "," +
```

Comment:
here can we assert formatted date like you have done in dateAddFunctionExtension7